api_v3/lib/types/entry/KalturaLiveEntry.php

```diff
@@ -274,4 +298,32 @@ protected function validateRecordingOptionsChanged($sourceObject, $attr)
 			$this->validateRecordingDone($sourceObject, "recordingOptions");
 		}
 	}
+
+	protected function validateSegmentDurationChanged($sourceObject, $attr)
+	{
+		$resolvedAttrName = $this->getObjectPropertyName($attr);
+		if(!$resolvedAttrName)
+			throw new KalturaAPIException(KalturaErrors::PROPERTY_IS_NOT_DEFINED, $attr, get_class($this));
+
+		/* @var $sourceObject LiveEntry */
+		$getter = "get" . ucfirst($resolvedAttrName);
+		if($sourceObject->$getter() !== $this->$attr) {
+			$this->validateSegmentDurationValue();
+		}
+	}
+
+	private function validateSegmentDurationValue()
+	{
+		$attrName = "segmentDuration";
```

Comment:
The attribute is passed as a parameter to the function (example validateRecordingOptionsChanged).
So use that format instead of the defining the name inside the function.